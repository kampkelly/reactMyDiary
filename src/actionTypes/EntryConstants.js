export const NEW_ENTRY = 'NEW_ENTRY';
