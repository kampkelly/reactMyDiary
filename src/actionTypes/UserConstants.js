export const SIGNIN = 'SIGNIN';
export const SIGNUP = 'SIGNUP';
export const SIGNOUT = 'SIGNOUT';
export const PROFILE = 'PROFILE';
export const NOTIFICATION = 'NOTIFICATION';