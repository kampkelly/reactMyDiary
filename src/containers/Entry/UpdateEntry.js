import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import './NewEntry.scss';
import '../../styles/form.scss';
import '../../styles/sidebar.scss';
import { ModifyEntry, ShowEntry } from '../../requests/EntryRequests';
import { validateForm } from '../../helpers/validateForm';
import icon from '../../assets/Rolling.svg';

/**
 * @class UpdateEntry
 * @extends {Component}
 */
class UpdateEntry extends Component {
  /**
   *Creates an instance of UpdateEntry.
   * @memberof NewEntry
   */
  constructor() {
    super();
    this.state = {
      description: '',
      entry: {},
      title: '',
      message: '',
      success: false
    };
    this.updateEntry = this.updateEntry.bind(this);
    this.change = this.change.bind(this);
  }

  componentDidMount() {
    this.props.ShowEntry(this.props.match.params.id);
    document.querySelector('body').insertAdjacentHTML('afterbegin', `<img src=${icon} id="loading"></img>`);
  }

  /**
   * @description - This method runs whenever redux state changes
   * @returns {object} state
   * @param {object} props
   * @param {object} state
   * @memberof ViewTag
   */
  static getDerivedStateFromProps(props, state) {
    return {
      entry: props.entry,
      message: props.message,
      success: props.success
    };
  }

  /**
   *
   * @returns {jsx} - jsx
   * @memberof NewEntry
   */
  updateEntry(e) {
    const errorMessage = 'One or more of the required fields are empty!';
    const values = { title: this.state.title, description: this.state.description };
    const noneEmpty = validateForm(errorMessage, values, e);
    if (noneEmpty === true) {
      document.querySelector('body').insertAdjacentHTML('afterbegin', `<img src=${icon} id="loading"></img>`);
      this.props.ModifyEntry(this.props.match.params.id, this.state.title, this.state.description);
    }
  }

  change(e) {
    this.setState({ [e.target.name]: e.target.value });
  }
  /**
   * @description - This method renders the jsx for this component
   * @returns {jsx} - jsx
   * @memberof Header
   */
  render() {
    const { title, description } = this.state.entry;
    return (
      <div>
        <section id="add">
          <div className="white-forms form-cover text-center">
            <form className="add_form" id="add-form">
              <h3 className="title text-center white-text">What is happening today?</h3>
              <p className="danger-text form_error_text"><small></small></p>
              <p className="success-text form_success_text"><small></small></p>
              <div className="field grid-container">
                <div className="col-2-6 col-6-6-md col-6-6-xs">
                  <label className="white-text">Title:</label>
                </div>
                <div className="col-4-6 col-6-6-md col-6-6-xs">
                  <input type="text" required="true" className="control-form" id="title" name="title" defaultValue={title} onChange={this.change} autoComplete="title" />
                </div>
              </div>
              <div className="field grid-container">
                <div className="col-2-6 col-6-6-md col-6-6-xs">
                  <label className="white-text">Description:</label>
                </div>
                <div className="col-4-6 col-6-6-md col-6-6-xs">
                  <textarea className="control-form" id="description" required="true" name="description" defaultValue={description} onChange={this.change} resize="no"></textarea>
                </div>
              </div>
              <div className="field grid-container">
                <div className="col-2-6 col-6-6-md col-6-6-xs"></div>
                <div className="col-2-6 col-6-6-md col-6-6-xs">
                  <button type="submit" className="submit-button button-white" onClick={this.updateEntry}>Update entry</button>
                </div>
                <div className="col-2-6 col-6-6-md col-6-6-xs cancel-div">
                  <button type="submit" className="submit-button button-cancel mt-4-xs mt-4-md">Cancel</button>
                </div>
                <div className="col-3-6 col-6-6-md col-6-6-xs">

                </div>
              </div>
          </form>
        </div>
      </section>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  entry: state.entry.entry,
  message: state.entry.message,
  success: state.entry.success
});

UpdateEntry.propTypes = {
  ModifyEntry: PropTypes.func,
  ShowEntry: PropTypes.func,
  success: PropTypes.bool
};

export default connect(mapStateToProps, {
  ModifyEntry, ShowEntry
})(UpdateEntry);

