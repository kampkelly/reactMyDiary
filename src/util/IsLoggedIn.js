export const isLoggedIn = () => {
  if (localStorage.getItem('diary_token') !== null) {
    return true;
  }
  return false;
};
