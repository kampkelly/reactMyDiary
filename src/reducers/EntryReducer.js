import { NEW_ENTRY, SHOW_ENTRY } from '../actionTypes/EntryConstants';
import { asyncActionName } from '../util/AsyncUtil';


const initialState = {
  entries: [],
  entry: {},
  loading: false,
  message: '',
  success: false,
  failure: false,
};

const entryReducer = (state = initialState, action = {}) => {
  switch (action.type) {
    case asyncActionName(NEW_ENTRY).loading:
      return {
        ...state,
        loading: action.payload,
      };
    case asyncActionName(NEW_ENTRY).success:
      return {
        ...state,
        success: action.payload
      };
    case asyncActionName(NEW_ENTRY).failure:
      document.getElementById('loading').style.display = 'none';
      document.querySelector('.form_error_text').style.display = 'block';
      document.querySelector('.form_error_text small').textContent = action.payload.error;
      return {
        ...state,
      };
    case asyncActionName(SHOW_ENTRY).loading:
      return {
        ...state,
        loading: action.payload,
      };
    case asyncActionName(SHOW_ENTRY).success:
      return {
        ...state,
        entry: action.payload, success: true
      };
    case asyncActionName(SHOW_ENTRY).failure:
      return {
        ...state, message: action.payload.error
      };
    default:
      return state;
  }
};

export default entryReducer;