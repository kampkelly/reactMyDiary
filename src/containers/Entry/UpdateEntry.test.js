import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import UpdateEntry from './UpdateEntry';
import EntryReducer from '../../reducers/EntryReducer';
import { asyncActions } from '../../util/AsyncUtil';
import { UPDATE_ENTRY } from '../../actionTypes/EntryConstants';

const mockStore = configureMockStore([thunk]);
const store = mockStore({
  entry: {
    entry: {},
    message: '',
    success: false
  }
});

const props = {
  history: {
    push: jest.fn()
  },
  match: {
    params: {
      id: 1
    }
  },
  ShowEntry: jest.fn()
};

let component;
let myComponent;

describe('<UpdateEntry/>', () => {
  beforeEach(() => {
    component = shallow(
      <Provider store={store}>
        <UpdateEntry {...props} />
      </Provider>
    );
    document.body.innerHTML =
    '<main></main>';
    myComponent = component.dive({ context: { store } }).dive();
  });
  it('should render without throwing an error', () => {
    expect(component).toMatchSnapshot();
  });
  it('should show update entry', () => {
    const payload = true;
    const expectedAction = {
      type: 'UPDATE_ENTRY_SUCCESS',
      payload
    };
    expect(asyncActions(UPDATE_ENTRY).success(payload)).toEqual(expectedAction);
  });
  it('should dispatch an action when updating an entry is loading', () => {
    const payload = false;
    const action = {
      type: 'UPDATE_ENTRY_LOADING',
      payload
    };
    const newState = EntryReducer({}, action);
    expect(newState).toEqual({
      loading: payload
    });
  });
  it('should dispatch an action to update an entry', () => {
    const payload = {};
    const action = {
      type: 'UPDATE_ENTRY_SUCCESS',
      payload
    };
    const newState = EntryReducer({}, action);
    expect(newState).toEqual({
      entry: {},
      success: true,
    });
  });
  it('should dispatch an action when updating an entry fails', () => {
    document.body.innerHTML =
    '<div>' +
    '  <span id="loading" />' +
    '  <span class="form_error_text" /><small></small><span>' +
    '  <button id="button" />' +
    '</div>';
    const payload = {
      error: ''
    };
    const action = {
      type: 'UPDATE_ENTRY_FAILING',
      payload
    };
    const newState = EntryReducer({}, action);
    expect(newState).toEqual({
      message: payload.error
    });
  });
  it('should click submit button', () => {
    document.body.innerHTML =
    '<body>' +
    '  <span id="loading" />' +
    '  <span class="form_error_text" /><small></small><span>' +
    '  <button id="button" />' +
    '</body>';
    const input = myComponent.find('input#title');
    input.simulate('change', { target: { value: 'New name' } });
    myComponent.find('button.button-white').simulate('click', {
      preventDefault: () => {
      }
     });
  });
  it('should click cancel button', () => {
    myComponent.find('button.button-cancel').simulate('click');
  });
});
