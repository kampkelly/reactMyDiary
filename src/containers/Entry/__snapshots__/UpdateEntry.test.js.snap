// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UpdateEntry/> should render without throwing an error 1`] = `
<Connect(UpdateEntry)
  ShowEntry={[MockFunction]}
  history={
    Object {
      "push": [MockFunction],
    }
  }
  match={
    Object {
      "params": Object {
        "id": 1,
      },
    }
  }
/>
`;
